### PR TITLE
Added convex hull class.

### PR DESCRIPTION
**Pull request by niklasborwell**

### CODE REVIEW

**Comment by niklasborwell** [comment]
Now a class.

File: source/math/FloatConvexHull2D.ooc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		newPoints := VectorList<FloatPoint2D> new(this count)
+		for (i in 0..this count)
+			newPoints add(transform * this _points[i])
+		This new(newPoints)
+	}
+	_isOnLeft: static func (leftPoint, rightPoint, queryPoint: FloatPoint2D) -> Bool {
+		(rightPoint x - leftPoint x) * (queryPoint y - leftPoint y) > (rightPoint y - leftPoint y) * (queryPoint x - leftPoint x)
+	}
+	free: func {
+		this _points free()
```
Comment (davidhesselbom) [inline source/math/FloatConvexHull2D.ooc]: You need to call `super` afterwards or the class instance itself will not be freed.

File: source/math/FloatConvexHull2D.ooc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		// TODO: Re-compute convex hull from new points if necessary, i.e. if transform changes order of points
+		newPoints := VectorList<FloatPoint2D> new(this count)
+		for (i in 0..this count)
+			newPoints add(transform * this _points[i])
+		This new(newPoints)
+	}
+	_isOnLeft: static func (leftPoint, rightPoint, queryPoint: FloatPoint2D) -> Bool {
+		(rightPoint x - leftPoint x) * (queryPoint y - leftPoint y) > (rightPoint y - leftPoint y) * (queryPoint x - leftPoint x)
+	}
+	free: func {
```
Comment (davidhesselbom) [inline source/math/FloatConvexHull2D.ooc]: Needs to be `override func`

**Comment by niklasborwell** [comment]
Updated.

**Comment by niklasborwell** [comment]
Rebased.

**Pull request merged**